### PR TITLE
Implement extensions and finish round trip tests

### LINKED ISSUE #7: Support SBOL extensions
Extensions are not currently supported. Extensions are used in a number of projects we know about. Add support for extensions.

### LINKED ISSUE #38: test_roundtrip has lots of differences in RDF
`test_roundtrip.py` compares the RDF after the roundtrip, and finds lots of differences. These are reported by default like this:

```
WARNING:sbol.test:Detected 24 differences in RDF
WARNING:sbol.test:Set environment variable SBOL_TEST_DEBUG to see details
```

To see the details run with the SBOL_TEST_DEBUG environment variable set (to any value), like this:

```shell
SBOL_TEST_DEBUG= python3 -m unittest sbol/test/test_roundtrip.py
```

### PR DESCRIPTION
**Pull request by tcmitchell**
Finish the work to implement extensions/annotations. And fix the last of the round trip tests that were not comparing properly.

Fixes #7
Fixes #38

### CODE REVIEW

File: sbol2/document.py [hunk 1]
```diff
@@ -525,6 +516,9 @@ def parse_objects_inner(self, subject, obj):
         if subject not in self.SBOLObjects and obj in self.SBOL_DATA_MODEL_REGISTER:
             # Call constructor for the appropriate SBOLObject
             new_obj = self.SBOL_DATA_MODEL_REGISTER[obj]()
+            if isinstance(new_obj, Identified):
+                # Clear out the version. it will get set later
+                new_obj.version = ''
```
Comment (bbartley) [inline sbol2/document.py:521]: I wondered if something like this might work...

**Comment by bbartley** [comment]
We may have to revisit #7 -- even though we can now round trip extensions, do we have everything in place to author extensions (i.e. extension properties, extension classes)?

**Comment by tcmitchell** [comment]
> We may have to revisit #7 -- even though we can now round trip extensions, do we have everything in place to author extensions (i.e. extension properties, extension classes)?

In the absence of further test cases, requirements, or examples, I think we must assume that extensions are complete. If/when gaps in the implementation are identified, issues should be opened to document the shortcomings, and to allow the gaps to be filled.

File: sbol2/document.py (hunk 1 above)
Comment (tcmitchell) [reply sbol2/document.py:521]: It is just one part of the recipe that allows versions to be properly handled.

**Pull request merged**